### PR TITLE
Adding a few more apub tests.

### PR DESCRIPTION
**Pull request by dessalines**
These are all the checks I could do from #1040, just through the API. The other ones will require http posting apub strings manually.

### CODE REVIEW

File: server/src/api/comment.rs [hunk 1]
```diff
@@ -151,7 +151,10 @@ impl Perform for CreateComment {
 
     // Check for a community ban
     let post_id = data.post_id;
-    let post = blocking(context.pool(), move |conn| Post::read(conn, post_id)).await??;
+    let post = match blocking(context.pool(), move |conn| Post::read(conn, post_id)).await? {
+      Ok(post) => post,
+      Err(_e) => return Err(APIError::err("couldnt_find_post").into()),
+    };
```
Comment (Nutomic) [inline server/src/api/comment.rs:157]: I would move this into a function get_post(), and also log the error (possibly at debug level).

**Comment by Nutomic** [comment]
Tests are passing fine when I'm running them locally, but it times out on Travis CI. No error message, maybe its running out of memory?
```
Creating travis_postgres_beta_1 ... 

Creating travis_postgres_gamma_1 ... 

Creating travis_iframely_1       ... 

Creating travis_pictrs_1         ... 

Creating travis_postgres_alpha_1 ... 

my-gamma_1    ... 

my-beta_1     ... 

my-alpha_1    ... 

myNet/lemmy/ui ~/build/LemmyNet/lemmy/docker/travis

Waiting for Lemmy to start...

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

**Comment by dessalines** [comment]
I see what's going on there. I forgot to update the travis docker-compose.

**Comment by dessalines** [comment]
Okay the tests passed now.

**Pull request merged**